### PR TITLE
Improved types (PHPStan level 6)

### PR DESCRIPTION
**Pull request by aivchen**
I had to disable 'void_return' php-cs-fixer rule. Otherwise it adds return type to `\PhpBench\Progress\LoggerInterface` after my changes.

### CODE REVIEW

**Comment by aivchen** [comment]
Bump

File: lib/Progress/Logger/BlinkenLogger.php [hunk 1]
```diff
@@ -177,9 +180,7 @@ private function drawIterations(Variant $variant, array $specials, $tag): void
         $line = sprintf('%-' . self::INDENT . 's', '#' . $variant->getSubject()->getIndex());
         $nbIterations = $variant->count();
 
-        for ($index = 0; $index < $nbIterations; $index++) {
-            $iteration = $variant->getIteration($index);
-
+        foreach ($variant as $index => $iteration) {
```
Comment (dantleech) [inline lib/Progress/Logger/BlinkenLogger.php:183]: Are we sure that index is a sequential list here? Maybe use array_values if not

Comment (aivchen) [reply lib/Progress/Logger/BlinkenLogger.php:183]: Yes, we are sure.
- In `lib/Model/Variant.php` we add values to the private property `Variant::$iterations` using only `$this->iterations[] = XXX`.
- `Variant::$iterations` has `/** @var list<Iteration> */` docblock. It means this array is 0-based sequential list (https://phpstan.org/writing-php-code/phpdoc-types#lists).

File: lib/Storage/HistoryEntry.php [hunk 2]
```diff
@@ -20,10 +20,38 @@
...
+        private                   $tag,
+        private                   $branch,
+        private                   $nbSubjects,
+        private                   $nbIterations,
+        private                   $nbRevolutions,
+        private                   $minTime,
+        private                   $maxTime,
+        private                   $meanTime,
+        private                   $meanRelStDev,
+        private                   $totalTime
```
Comment (dantleech) [inline lib/Storage/HistoryEntry.php:48]: :smile:

**Comment by dantleech** [comment]
great work again :bow:

File: lib/Progress/Logger/BlinkenLogger.php (hunk 1 above)
Comment (dantleech) [reply lib/Progress/Logger/BlinkenLogger.php:183]: yes, have some ancient memories about making for loops like that to work around "lists" that had been filtered etc making their array keys non-sequential.

**Pull request merged**